tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
-            bucket_name, object_names[i], get_upload_id, all_part_info
+            resp_dir[f"{object_names[i]}_upload_id"] = get_upload_id
+            all_part_info = []
+            file_name = origin_dir + "/" + object_names[i]
+            part_size = "10M"
+            log.info(f"Split data into {part_size} size")
+            part_data = split_file_data_for_multipart_upload(file_name, part_size)
+            log.info("Initiate part uploads for multipart object")
+            for pd in range(len(part_data)):
+                part_id = pd + 1
+                part_info = c_scope_s3client.initiate_upload_part(
+                    bucket_name,
+                    object_names[i],
+                    part_id,
+                    get_upload_id,
+                    part_data[pd],
+                )
+                all_part_info.append({"PartNumber": part_id, "ETag": part_info["ETag"]})
+            resp_dir["all_part_info"] = all_part_info
+        return resp_dir
+
+    def test_s3_multipart_upload(
+        self,
+        c_scope_s3client,
+        tmp_directories_factory,
+    ):
+        """
+
+        Test basic s3 operations using a noobaa bucket:
+        1. Create an account
+        2. Create a bucket using S3
+        3. Write multipart objects to the bucket
+        4. List multipart objects from the bucket
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
         )
         assert (
             mp_response["ResponseMetadata"]["HTTPStatusCode"] == 200
         ), "Failed to upload multipart object"
         log.info(mp_response)
 
-    # 4. List multipart objects from the bucket
-    listed_objs = s3_client.list_objects(bucket_name)
-    assert set(object_names).issubset(
-        set(listed_objs)
-    ), "All uploaded objects are not present in bucket"
+    def test_s3_multipart_download(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test basic s3 operations using a noobaa bucket:
+        1. Write multipart objects to the bucket
+        2. Download the objects from the bucket and verify data integrity
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
```

Comment:
f"{obj_name}" is equivalent to obj_name - see the other comment for suggestoins

Reply:
Fixed